core/iwasm/common/wasm_memory.c

```diff
@@ -16,6 +16,7 @@ typedef enum Memory_Mode {
 } Memory_Mode;
 
 static Memory_Mode memory_mode = MEMORY_MODE_UNKNOWN;
+static korp_mutex mem_mode_lock;
```

Comment:
could we make `memory_mode` atomic instead?

Reply:
right, that's better since `memory_mode` is a simple enum